src/sinks/aws_kinesis_firehose/tests.rs

```diff
@@ -17,50 +17,43 @@ fn generate_config() {
 
 #[tokio::test]
 async fn check_batch_size() {
+    // Sink builder should limit the batch size to the upper bound.
+    let mut batch = BatchConfig::<KinesisFirehoseDefaultBatchSettings>::default();
+    batch.max_bytes = Some(MAX_PAYLOAD_SIZE + 1);
+
     let config = KinesisFirehoseSinkConfig {
         stream_name: String::from("test"),
         region: RegionOrEndpoint::with_endpoint("http://localhost:4566".into()),
         encoding: EncodingConfig::from(StandardEncodings::Json),
         compression: Compression::None,
-        batch: BatchConfig {
-            max_bytes: Some(MAX_PAYLOAD_SIZE + 1),
-            ..Default::default()
-        },
+        batch,
         request: Default::default(),
         assume_role: None,
         auth: Default::default(),
     };
 
     let cx = SinkContext::new_test();
     let res = config.build(cx).await;
-
-    assert_eq!(
-        res.err().and_then(|e| e.downcast::<BuildError>().ok()),
-        Some(Box::new(BuildError::BatchMaxSize))
-    );
+    assert!(res.is_ok());
```

Comment:
So, the error was being emitted before because the sink wanted to emit an error if the max bytes/max events was exceeded.  I changed the logic to simply cap the values.

It's definitely a behavior change, but it felt like if the intention is to force people to adjust their configurations to observe the limit, then enforcing the limit silently would be less toil for the user if they fatfingered the wrong number or something.

Thoughts?  I'm happy to change it back.

Reply:
I'm open to other thoughts as well. I just think, as a user, if I configured a batch size of 100 MB and it sent batches of 10 MB I would be very confused. We could document it, but I think error'ing makes the cap clearly visible.

Reply:
Yeah, I'm down to change it.

Reply:
I updated the builder methods used for limiting, etc, to return an error if the limit is exceeded, so we'll propagate those errors up to the caller in all situations now.